src/rns/rns.blockchain.ts

```diff
@@ -73,67 +75,85 @@ async function nameChangedHandler (eventData: EventData): Promise<void> {
   }
 }
 
-async function updatePlacementHandler (eventData: EventData): Promise<void> {
-  // UpdatePlacement(tokenId, paymentToken, cost)
+async function tokenPlacedHandler (eventData: EventData): Promise<void> {
+  // event TokenPlaced(uint256 indexed tokenId, address indexed paymentToken, uint256 cost);
+
   const transactionHash = eventData.transactionHash
   const tokenId = Utils.numberToHex(eventData.returnValues.tokenId)
   const paymentToken = eventData.returnValues.paymentToken
   const cost = eventData.returnValues.cost
 
-  if (cost === '0') {
-    // Canceled or sold
-    logger.info(`Canceled or sold: ${tokenId}`)
-    const lastOffer = await DomainOffer.findOne({ where: { tokenId: tokenId, status: 'ACTIVE' } })
+  const domain = await Domain.findByPk(tokenId)
 
-    if (lastOffer) {
-      logger.info(`Found last offer for ${tokenId}`)
-      lastOffer.status = 'CANCELED'
-      lastOffer.save()
+  if (!domain) {
+    throw new Error(`Domain with token ID ${tokenId} not found!`)
+  }
 
-      const [affectedRows, realAffectedRows] = await SoldDomain.update({
-        price: lastOffer.price,
-        paymentToken: lastOffer.paymentToken,
-        soldDate: Date.now()
-      }, { where: { id: transactionHash } })
+  const domainOffer = await DomainOffer.create({
+    offerId: transactionHash,
+    sellerAddress: domain.ownerAddress,
+    tokenId: tokenId,
+    paymentToken: paymentToken,
+    price: cost,
+    creationDate: Date.now(), // TODO: get from block timestamp
+    status: 'ACTIVE'
+  })
+
+  if (domainOffer) {
+    logger.info(`TokenPlaced event: ${tokenId} created`)
+  }
+}
 
-      if (affectedRows) {
-        logger.info(`UpdatePlacement event: Sold Domain ${tokenId}`)
-      } else {
-        logger.info(`UpdatePlacement event: no Domain ${tokenId} updated`)
-      }
-    }
+async function tokenUnplacedHandler (eventData: EventData): Promise<void> {
+  // event TokenUnplaced(uint256 indexed tokenId);
+
+  const tokenId = Utils.numberToHex(eventData.returnValues.tokenId)
+
+  const [affectedRows, realAffectedRows] = await DomainOffer.update({
+    status: 'CANCELED'
+  }, { where: { tokenId: tokenId, status: 'ACTIVE' } })
+
+  if (affectedRows) {
+    logger.info(`TokenUnplaced event: ${tokenId} updated`)
   } else {
-    const price = cost
+    logger.info(`TokenUnplaced event: ${tokenId} not updated`)
+  }
+}
 
-    const domain = await Domain.findByPk(tokenId)
+async function tokenSoldHandler (eventData: EventData): Promise<void> {
+  // event TokenSold(uint256 indexed tokenId);
 
-    if (!domain) {
-      throw new Error(`Domain with token ID ${tokenId} not found!`)
-    }
+  const transactionHash = eventData.transactionHash
+  const tokenId = Utils.numberToHex(eventData.returnValues.tokenId)
+
+  const lastOffer = await DomainOffer.findOne({ where: { tokenId: tokenId, status: 'ACTIVE' } })
+
+  if (lastOffer) {
+    logger.info(`Found last offer for ${tokenId}`)
+    lastOffer.status = 'CANCELED'
```

Comment:
Shouldn't there be different status than `CANCELED` for sold domains?

Reply:
have a good eye, I tried to reflect the Placement contract behavior and to avoid duplicate the sold state